### PR TITLE
Codes of Real-Time IoT Perception Systems Based on Edge-Cloud Collaboration with Large Foundation Models

### PR DESCRIPTION
**Pull request by hsj576**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR provides codes of Real-Time IoT Perception Systems Based on Edge-Cloud Collaboration with Large Foundation Models.

**Which issue(s) this PR fixes**:

Fixes #80

### CODE REVIEW

**Comment by MooreZheng** [comment]
After the presentation in the routine meeting, the overall result looks fine to us. 

The project would become even better if 
1. Might combine the OSPP project with the LFX one, since they are using the same dataset, with the same scheme, and under the same task. 
2. Add comments, APIs, and tutorials for the project. 
3. Add docs of scenarios, algorithms, leaderboards, and test reports on https://ianvs.readthedocs.io/en/latest/

**Comment by hsj576** [comment]
> After the presentation in the routine meeting, the overall result looks fine to us.
> 
> The project would become even better if
> 
> 1. Might combine the OSPP project with the LFX one, since they are using the same dataset, with the same scheme, and under the same task.
> 2. Add comments, APIs, and tutorials for the project.
> 3. Add docs of scenarios, algorithms, leaderboards, and test reports on https://ianvs.readthedocs.io/en/latest/

I made the following modifications: 
1. Merged the code for the OSPP project and the LFX project, now they are both located under examples/robot. 
2. Added test reports related to the Cloud-Robotics dataset under docs/leaderboards. 
3. Added comments for the APIs, code interfaces in the form of python docstrings.

File: examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        """
+        if len(data) > 10:
+            print("predict start for big data")
+            my_kwargs = {'num_workers': self.val_args.workers, 'pin_memory': True}
+            _, _, self.validator.test_loader, _ = make_data_loader(self.val_args, test_data=data, **my_kwargs)
+        else:
+            print("predict start for small data")
+            if not isinstance(data[0][0], dict):
+                data = self._preprocess(data)
+                #print("predict starting 69")
```
Comment (jaypume) [inline examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py:96]: The lines for local debug should not be committed, these TEMP PRINT lines should be deleted.

File: examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            if type(data) is np.ndarray:
+                data = data.tolist()
+                #print("predict starting 72")
+            #print("predict starting 73")
+            self.validator.test_loader = DataLoader(data, batch_size=self.val_args.test_batch_size, shuffle=False,
+                                                pin_memory=True)
+        
+        #print("predict starting 75")
```
Comment (jaypume) [inline examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py:104]: All these TEMP PRINT lines should be deleted

File: examples/robot/lifelong_learning_bench/testalgorithms/rfnet/RFNet/utils/args.py [hunk 3]
```diff
@@ -21,7 +21,7 @@ def __init__(self, **kwargs):
         self.gpu_ids = 0
 
         self.seed = 1
-        self.resume = None
+        self.resume = '/home/hsj/ianvs/project/pretrain/pretrain_model.pth'
```
[the remark this answers is not in the document]
Comment (hsj576) [reply examples/robot/lifelong_learning_bench/testalgorithms/rfnet/RFNet/utils/args.py]: The hardcode path is removed.

File: examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    def set_weights(self, weights):
+        self.trainer.set_weight(weights)
+        
+        epoch_num = 0
+        print("Total epoch: ", epoch_num)
+        loss_all = []
+        for epoch in range(epoch_num):
+            train_loss = self.trainer.my_training(epoch)
+            loss_all.append(train_loss)
+        with open('/home/shijing.hu/ianvs/project/ianvs/train_loss_2.txt', 'a+') as file:
```
[the remark this answers is not in the document]
Comment (hsj576) [reply examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py]: The hardcode path is removed.

File: examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py (hunk 1 above)
Comment (hsj576) [reply examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py]: TEMP PRINT lines are removed.

File: examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py (hunk 2 above)
Comment (hsj576) [reply examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py]: TEMP PRINT lines are deleted.

**Comment by jaypume** [comment]
Please squash these commits to make commit history more clean, and the better count should be 3-5 commits.
You can just do it by squashing the duplicate commits.

**Comment by hsj576** [comment]
> Please squash these commits to make commit history more clean, and the better count should be 3-5 commits. You can just do it by squashing the duplicate commits.

The duplicate commits are squashed.

**Pull request merged**